### DISCUSSION TITLE
How to enable TOC for all headers

### DISCUSSION THREAD

**Discussion by opabravo** [cotes2020/jekyll-theme-chirpy#1364] [Q&A] [answered]
### Checklist

- [X] I have read and accepted the [contributing guidelines](https://github.com/cotes2020/jekyll-theme-chirpy/blob/master/docs/CONTRIBUTING.md).

### How did you create the site?

Generated from `chirpy-starter`

### Description

Is there a way to enable all h2~h6 headeres in TOC section?

### Operations you have already tried

- Read through the doc
- Google

**Comment by NichtsHsu** [comment]
https://github.com/cotes2020/jekyll-theme-chirpy/blob/5f2edb0914c0bc2254c6802d615f22cdb5b79dce/_javascript/modules/components/toc.js#L8

**Reply by opabravo** [reply]
Hi, thanks for replying,

Since i'm using the `starter` repo, I can't find the script to edit,

would you like to provide more detailed guidance/refer?

**Comment by kungfux** [comment]
@opabravo  In case of `starter`, there is no way to override javascript files, it seems. However, we can add our own. So, here is the solution for blogs created by `starter`:

Create a file `assets\js\custom-toc.js` with the [configuration you want](https://github.com/tscanlin/tocbot?tab=readme-ov-file#options):

```javascript
!function(){
    "use strict";
    document.querySelector("main h2, main h3")&&tocbot.init({
        tocSelector:"#toc",
        contentSelector:".content",
        ignoreSelector:"[data-toc-skip]",
        headingSelector:"h2, h3, h4, h5, h6",
        orderedList:!1,
        scrollSmooth:!1
    })}();
```

Since custom scripts are being executed later, it overrides configuration successfully. However, I'm not sure this will work stable for 100%, but it seems to work.

**Reply by huanyushi** [reply]
@kungfux Perhaps I didn't make myself clear earlier. Please allow me to restate it.

> In case of starter, there is no way to override javascript files, it seems.

I have created a file `_javascript\modules\components\toc.js`, it can not override the theme's javascript file, as you said.

> Create a file `assets\js\custom-toc.js` with the [configuration you want](https://github.com/tscanlin/tocbot?tab=readme-ov-file#options):

So I followed the solution you mentioned later. I create a new file `assets\js\custom-toc.js`, like this:
```js
!function(){
    "use strict";
    document.querySelector("main h2, main h3")&&tocbot.init({
        tocSelector:"#toc",
        contentSelector:".content",
        ignoreSelector:"[data-toc-skip]",
        headingSelector:"h2, h3, h4, h5, h6",
        orderedList:!1,
        scrollSmooth:!1,
        collapseDepth:6,
    })}();
```
I want the post to display all TOC without animation. After creating this file, and run `bundle exec jekyll s`, the TOC does not change. 

So it seems like this method isn't effective, or perhaps I missed something. Do I need to insert `<script src="{{ site.baseurl }}/assets/js/custom-toc.js"></script>`  into some file? I have no idea.

Could you provide more specific details? I'm not familiar with javascript, any suggestions would be greatly appreciated.

Relevent link: #1616

**Reply by kungfux** [reply]
@huanyushi I gave it a try and it didn't work for me either anymore. I found no changes that may break it, so I have no ideas to share at the moment.

**Reply by huanyushi** [reply]
@kungfux Thanks for your reply sincerely, maybe I should create a new discussion later.

**Comment by kungfux** [comment, marked as answer]
Here is the working solution.

Create a file `_includes/metadata-hook.html` with the following content (this file is completely empty in Chirpy repo and is made specifically to include extra information into the `<head>` section without the need to modify and support other files like `head.html`):

```html
<script type="text/javascript" src="/assets/js/toc.js" defer></script>
```

Then, create a javascript file `assets/js/toc.js` to overrides TOC configuration e.g.:

```js
document.querySelector("main h2, main h3")&&tocbot.refresh({
    tocSelector:"#toc",
    contentSelector:".content",
    ignoreSelector:"[data-toc-skip]",
    headingSelector:"h2, h3, h4, h5, h6",
    orderedList:!1,
    scrollSmooth:!1,
    collapseDepth: 6
});
```

Here is [tocbot documentation](https://github.com/tscanlin/tocbot#options) regarding available options.

FYI @huanyushi @opabravo

**Reply by TajangSec** [reply]
Currently invalid
